### PR TITLE
Add auto-sni support

### PR DESCRIPTION
**Pull request by kfaseela**
Signed-off-by: Faseela K <faseela.k@est.tech>

**Please provide a description of this PR:**

[RFC auto-sni support](https://docs.google.com/document/d/1pTUl-Ng3nXAWJb7UGJtalftznpxQEfID/edit#)

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: releasenotes/notes/auto-sni-support.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: traffic-management
+releaseNotes:
+   - |
+     **Added** Auto-Sni Support.
```
Comment (jacob-delgado) [inline releasenotes/notes/auto-sni-support.yaml:6]:
```suggestion
          **Added** the ability to configure automatically set SNI when `DestinationRules` do not specify it and `VERIFY_CLIENT_AT_CERT` is enabled.
```

Comment (kfaseela) [reply releasenotes/notes/auto-sni-support.yaml]: Done

**Review by jacob-delgado — APPROVED** [review]

**Comment by Kmoneal** [comment]
LGTM

**Pull request merged**